### ISSUE TITLE
samtools mpileup is fixed and we no longer need to add `*` in zero coverage regions

### ISSUE THREAD

**Issue by mfoll** [IARCbioinfo/needlestack#101]
We can remove the `| sed 's/        /   *   */g'` from here https://github.com/IARCbioinfo/needlestack/blob/v0.3b/needlestack.nf#L236 since samtools mpileup bug has been fixed:
https://github.com/samtools/samtools/commit/afae50c6eb58298873f6768ed7c49fd4fb0b3438

It's not yet present in samtools 1.3 so we should wait until the next release to do it. 

This will solved an unexpected behaviour in needlestack: because of the pipe, even if samtools returns a non-zero exit code, the command returns a zero exit code (see http://unix.stackexchange.com/questions/14270/get-exit-status-of-process-thats-piped-to-another). One option would be to use `pipefail` in bash but it's not compatible with all bash versions and also doesn't work will with `head -n 1`like here https://github.com/IARCbioinfo/needlestack/blob/v0.3b/needlestack.nf#L276. For example:

``` bash
$ seq 1 100000 | head -1 ; echo $?
1
0
$ set -o pipefail ; seq 1 100000 | head -1 ; echo $?
1
141
```

**Comment by mfoll** [comment]
Fixed in samtools 1.3.1

**Comment by mfoll** [comment]
Note that this has been fixed since samtools release 1.4: https://github.com/samtools/samtools/releases/tag/1.4
> The mpileup text output no longer contains empty columns at zero coverage
  positions.  Previously it would output "...0\t\t..." in some circumstances
  (zero coverage due to being below a minumum base quality); this has been
  fixed to output as "...0\t*\t*..." with placeholder '*' characters as in
  other zero coverage circumstances (see PR #537).

**Comment by mfoll** [comment]
It doesn't hurt to keep the sed command to keep compatibility with old version of samtools, so keeping this for the moment.
